compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -417,6 +417,9 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 // These just return their argument
                 self.copy_op(&args[0], dest)?;
             }
+            sym::cold_path => {
+                // This is a no-op. The intrinsic is just a hint to the optimizer.
+            }
```

Comment:
You added a fallback impl, so this should not be needed.

Reply:
Same for the cranelift impl.

Reply:
That's intentional to avoid emitting a call to the fallback impl which is an empty function.

Reply:
Then why do we have a fallback impl?

Reply:
It's used by miri.

Reply:
No it is not. Miri uses this implementation I attached the comment to.

Reply:
https://github.com/rust-lang/rust/pull/133163 cleans this up